pallets/gear/src/lib.rs

```diff
@@ -699,8 +651,13 @@ pub mod pallet {
                 T::DebugInfo::remap_id();
             }
 
-            while Self::can_process_queue() {
-                if let Some(dispatch) = common::dequeue_dispatch() {
+            while <MessengerPallet<T> as Messenger>::QueueProcessing::allowed() {
+                if let Some(dispatch) = <MessengerPallet<T> as Messenger>::Queue::pop_front()
+                    .unwrap_or_else(|_| {
+                        // Can be called only in case of storage corruption
```

Comment:
> All these `unreachables` make me sad. Can we eliminate them somehow?

They are seems logical for me, because this is the end of some algorithmic invariants, which are unreachable if we don't force some bad mutates to the storage

Reply:
> maybe a bit more details about that? You can write a "proof" in one place and refer to it

I would like to migrate anything and provide big readme with explanations how all pallets works some time soon

Reply:
#899